packages/aws-lambda-otel-extension/opt/otel-extension/external/helper.js

```diff
@@ -10,6 +10,39 @@ const extensionVersion = (() => {
   return require('../version');
 })();
 
+const cleanResponseData = (data) => {
+  let response = '';
+  data.excludedBody = false;
+
+  let responseData = data.responseData;
+  if (typeof responseData === 'string') {
+    try {
+      responseData = JSON.parse(responseData);
+    } catch (error) {
+      responseData = data.responseData;
+    }
+  }
+
+  // Only save responses if it is an object
+  if (responseData && typeof responseData === 'object') {
+    if ('body' in responseData && typeof responseData.body === 'string') {
+      try {
+        JSON.parse(responseData.body);
```

Comment:
What's the purpose of this line?

If this logic is meant to be run only if we parsed `responseData` out of JSON. It'll be good to form it more explicitly. In given form, it's hard to get this intention

Reply:
This line is meant to check if the body of an http request is JSON or not.

At this point in the check the `responseData` would look something like this

```json
{
    "traceId": "xxxx",
    "spanId": "xxxxx",
    "executionId": "xxx-xxxx-xxxx-xxxx-xxxxxx",
    "orgUid": "xxxxx",
    "requestData": "",
    "responseData": "{\"statusCode\":200,\"headers\":{\"access-control-allow-origin\":\"*\",\"content-type\":\"application/json; charset=utf-8\"},\"body\":\"{\\\"message\\\":\\\"Success\\\"}}\"}",
    "timestamp": "2022-04-27T19:29:47.151000022Z"
  }
```

And we would want to check that the `body` inside the `responseData` is JSON or not. If it is not JSON we would want to drop the body.

Reply:
@medikoo I refactored this a bit to make this more clear. Let me know what you think 👍